app/org/sagebionetworks/bridge/dynamodb/DynamoSubpopulationDao.java

```diff
@@ -164,9 +191,36 @@ public void deleteAllSubpopulations(StudyIdentifier studyId) {
         if (!subpops.isEmpty()) {
             for (Subpopulation subpop : subpops) {
                 studyConsentDao.deleteAllConsents(subpop.getGuid());
+                criteriaDao.deleteCriteria(subpop.getKey());
             }
             List<FailedBatch> failures = mapper.batchDelete(subpops);
             BridgeUtils.ifFailuresThrowException(failures);
         }
     }
+    
+    private String getKey(Subpopulation subpop) {
+        return "subpopulation:" + subpop.getGuidString();
+    }
+    
+    // Save the criteria object if it exists. If it does not, copy the criteria data from the subpopulation
+    // and save that. Set that on subpopulation. Once all models have a criteria object, this will be removed 
+    // along with the fields on subpopulation.
+    private void persistCriteria(Subpopulation subpop) {
+        Criteria criteria = subpop.getCriteria();
+        Criteria makeCopyOf = (criteria == null) ? subpop : criteria;
+
+        Criteria copy = criteriaDao.copyCriteria(getKey(subpop), makeCopyOf);
+        criteriaDao.createOrUpdateCriteria(copy);
+        subpop.setCriteria(copy);
+    }
```

Comment:
If we refactor such that null criteria means match everything and manually migrate the 2 non-trivial criteria, we could simplify or remove the back-compat logic here.